### PR TITLE
PR: Keep finfo.newly_created state when cancelling save_as

### LINKED ISSUE #11058: Exceptions when I cancelled saving an unsaved file
### What steps will reproduce the problem?

Not sure how to reproduce exactly. I was editing and running a script repeatedly without saving it. I hit CTRL+S, and then in the Windows save dialog box I just hit cancel. I had done that a few times already with no problem. This time after 2-3 seconds this issue reporter automatically popped up.

### Traceback
```python-traceback
  File "C:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\editor\utils\autosave.py", line 116, in do_autosave
    stack.autosave.autosave_all()
  File "C:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\editor\utils\autosave.py", line 375, in autosave_all
    self.maybe_autosave(index)
  File "C:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\editor\utils\autosave.py", line 335, in maybe_autosave
    orig_hash = self.file_hashes[orig_filename]
KeyError: 'C:\\dev\\trunk\\applications\\GreenSideGolfSolution\\Python\\TrackAnalysis\\untitled1.py'
WARNING:spyder.widgets.github.backend:Failed to send bug report on Github. response={'code': 401, 'json': {'message': 'Bad credentials', 'documentation_url': 'https://developer.github.com/v3'}}
WARNING:spyder.widgets.github.backend:Failed to send bug report on Github. response={'code': 401, 'json': {'message': 'Bad credentials', 'documentation_url': 'https://developer.github.com/v3'}}
```

## Versions

* Spyder version: 4.0.0 
* Python version: 3.7.3
* Qt version: 5.9.6
* PyQt5 version: 5.9.2
* Operating System: Windows 7

### Dependencies

```
cloudpickle >=0.5.0          :  1.2.2 (OK)
pygments >=2.0               :  2.5.2 (OK)
qtconsole >=4.6.0            :  4.6.0 (OK)
nbconvert >=4.0              :  5.6.1 (OK)
sphinx >=0.6.6               :  2.2.2 (OK)
pylint >=0.25                :  2.4.4 (OK)
psutil >=0.3                 :  5.6.7 (OK)
qtawesome >=0.5.7            :  0.6.0 (OK)
qtpy >=1.5.0                 :  1.9.0 (OK)
pickleshare >=0.4            :  0.7.5 (OK)
zmq >=17                     :  18.1.0 (OK)
chardet >=2.0.0              :  3.0.4 (OK)
numpydoc >=0.6.0             :  0.9.1 (OK)
spyder_kernels >=1.8.1;<2.0.0:  1.8.1 (OK)
qdarkstyle >=2.7             :  2.7 (OK)
atomicwrites >=1.2.0         :  1.3.0 (OK)
diff_match_patch >=20181111  :  20181111 (OK)
intervaltree                 :  None (OK)
watchdog                     :  None (OK)
keyring                      :  None (OK)
pexpect >=4.4.0              :  4.7.0 (OK)
pympler                      :  None (OK)
sympy >=0.7.3                :  1.4 (OK)
cython >=0.21                :  0.29.14 (OK)
IPython >=4.0                :  7.10.1 (OK)
matplotlib >=2.0.0           :  3.1.1 (OK)
pandas >=0.13.1              :  0.25.3 (OK)
numpy >=1.7                  :  1.17.4 (OK)
scipy >=0.17.0               :  1.3.2 (OK)
pyls >=0.31.2;<0.32.0        :  0.31.2 (OK)
rtree >=0.8.3                :  0.8.3 (OK)
```

### PR DESCRIPTION
**Pull request by impact27**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Issue(s) Resolved

Fixes #11058

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

### CODE REVIEW

File: spyder/plugins/editor/widgets/editor.py [hunk 1]
```diff
@@ -2078,6 +2078,7 @@ def save_as(self, index=None):
         # The next line is necessary to avoid checking if the file exists
         # While running __check_file_status
         # See spyder-ide/spyder#3678 and spyder-ide/spyder#3026.
+        original_newly_created = finfo.newly_created
```
Comment (jitseniesen) [inline spyder/plugins/editor/widgets/editor.py]: Perhaps you can reformulate the comment above, because "next line" now refers to the new line you added.

**Comment by ccordoba12** [comment]
@jitseniesen, please approve this PR when you consider it's ready.

**Comment by jitseniesen** [comment]
The code looks okay and straightforward but before approving I want to check that is does indeed work. I will try to do this tonight, or tomorrow if I cannot make it before.

**Comment by ccordoba12** [comment]
Thanks @jitseniesen!

**Pull request merged**